chainio/clients/avsregistry/reader.go

```diff
@@ -429,7 +429,7 @@ func (r *ChainReader) QueryExistingRegisteredOperatorPubKeys(
 
 	operatorAddresses := make([]types.OperatorAddr, 0)
 	operatorPubkeys := make([]types.OperatorPubkeys, 0)
-	for i := startBlock; i.Cmp(stopBlock) <= 0; i.Add(i, blockRange) {
+	for i := new(big.Int).Set(startBlock); i.Cmp(stopBlock) <= 0; i.Add(i, blockRange) {
```

Comment:
```suggestion
        // QueryExistingRegisteredOperatorPubKeys and QueryExistingRegisteredOperatorSockets both run in parallel and they read and mutate the same variable startBlock, so we clone it to prevent the race condition
	for i := new(big.Int).Set(startBlock); i.Cmp(stopBlock) <= 0; i.Add(i, blockRange) {
```

Reply:
Done!